samples/spring-security-basic-auth/src/main/java/sample/spring/xsuaa/security/TokenBrokerResolver.java

```diff
@@ -0,0 +1,126 @@
+/**
+ * SPDX-FileCopyrightText: 2018-2023 SAP SE or an SAP affiliate company and Cloud Security Client Java contributors
+ * <p>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package sample.spring.xsuaa.security;
+
+import com.sap.cloud.security.xsuaa.tokenflows.PasswordTokenFlow;
+import com.sap.cloud.security.xsuaa.tokenflows.TokenFlowException;
+import com.sap.cloud.security.xsuaa.tokenflows.XsuaaTokenFlows;
+import jakarta.servlet.http.HttpServletRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.cache.Cache;
+import org.springframework.http.HttpHeaders;
+import org.springframework.lang.Nullable;
+import org.springframework.security.oauth2.server.resource.web.BearerTokenResolver;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+import java.util.Optional;
+
+/**
+ * Resolves with an XSUAA user token from the configured service instance using the Basic HTTP credentials
```

Comment:
how about:
Resolves an user token from the configured XSUAA service instance via password grant flow using the Basic HTTP credentials from the request header.